core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -216,7 +216,9 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
       if (driverConf.contains("spark.yarn.credentials.file")) {
         logInfo("Will periodically update credentials from: " +
           driverConf.get("spark.yarn.credentials.file"))
-        SparkHadoopUtil.get.startCredentialUpdater(driverConf)
+        Utils.classForName("org.apache.spark.deploy.yarn.YarnSparkHadoopUtil")
```

Comment:
What's the purpose of changing to this way?

Reply:
`YarnSparkHadoopUtil` does not extend `SparkHadoopUtil` anymore (it doesn't even exist as a class), so I'm using this to invoke the method on the YARN object.

I plan to clean this up once #19272 is in, since that's a cleaner way to do this token renewal thing.

Reply:
I see, thanks for explanation, this kind of reflection seems not so elegant.

Reply:
#19272 is already in, do you plan to update it here in this PR or in a separate PR?

Reply:
I plan to do it in a separate PR (also to make YARN and Mesos share more code in that area).